spark/src/main/java/org/apache/iceberg/actions/BaseSparkAction.java

```diff
@@ -64,6 +70,20 @@ protected JavaSparkContext sparkContext() {
     return sparkContext;
   }
 
+  protected <T> T withJobGroupInfo(JobGroupInfo info, Supplier<T> supplier) {
+    SparkContext context = spark().sparkContext();
+    JobGroupInfo previousInfo = JobGroupUtils.getJobGroupInfo(context);
+    try {
+      JobGroupUtils.setJobGroupInfo(context, info);
+      return supplier.get();
+    } finally {
+      JobGroupUtils.setJobGroupInfo(context, previousInfo);
+    }
+  }
+
+  protected JobGroupInfo newJobGroupInfo(String groupId, String desc) {
+    return new JobGroupInfo(groupId, desc + "-" + jobCounter.incrementAndGet(), false);
```

Comment:
I think I made a typo in the snippet. We should probably append jobCounter to id, not desc. My bad.